common/oauth.go

```diff
@@ -40,6 +41,7 @@ type oauthClientParams struct {
 	token       *oauth2.Token
 	config      *oauth2.Config
 	tokenSource oauth2.TokenSource
+	updateToken func(tok *oauth2.Token) error
```

Comment:
Would `onTokenUpdate` be a better name for this? Since this function itself is not updating the token, it is the callback function for when the token is updated.

Reply:
`tokenUpdated`? The `on` prefix is something I'd usually call a parameter exposed to the caller, but this doesn't get exposed to the caller.